lib/utils.py

```diff
@@ -57,43 +57,47 @@ def make_request(
     url = BASE_URL + site
 
     attempts = 0
-    while attempts < max_attempts:
-        attempts += 1
-        if method == "POST":
-            response = requests.post(url, headers=headers, json=info)
-        else:
-            response = requests.get(url, headers=headers, params=info)
-
-        if response.status_code == 200:
-            logger.debug("Successfully made request after %d attempts", attempts)
-            return response.json()
-
-        # Request did not succeed
-        response_body = response.content.decode()
-        error_msg = response.reason + " " + str(response.status_code)
-        error = RequestError(error_msg, response_body)
-
-        try:
-            _handle_southwest_error_code(error)
-        except (RequestError, AirportCheckInError) as err:
-            # Stop requesting after one attempt for special codes, as the requests won't succeed
-            error = err
-            break
-
-        if random_sleep:
-            sleep_time = random_sleep_duration(1, 3)
-        else:
-            sleep_time = 0.5
-
-        logger.debug(
-            f"Request error on attempt {attempts}: {error_msg}. Sleeping for {sleep_time:.2f} "
-            "seconds until next attempt"
-        )
-        time.sleep(sleep_time)
-
-    logger.debug("Failed to make request after %d attempts: %s", attempts, error_msg)
-    logger.debug("Response body: %s", response_body)
-    raise error
+    session = requests.Session()
```

Comment:
What’s the reason for using a session here? If we hit a 403 for a few requests, I would think we wouldn’t want to keep certain parameters attached to subsequent requests as it could make it easier to detect the script (cookies were used for the first request -> 403. Same cookies are used again, so probably reject again because last time was a 403). It’s obviously not clear how exactly they detect, so those are just my speculations.

Reply:
I used session because I was getting out of memory error locally - I don't recall the exact error but using session and closing it after each request resolved the error. Regarding cookie I have removed it because it does not show in the headers anyway. Related to detection, I have concluded is that either wrong user-agent or IP address (VPN) - this is how Southwest detects it. Since we are using is_mobile so user agent is same for everyone so now IP address is the main concern as it should be residential and not flagged by cloudflare...

Reply:
Could you be a bit more specific on the issue you were receiving? This may be a separate issue.

> Regarding cookie I have removed it because it does not show in the headers anyway.

[Sessions](https://docs.python-requests.org/en/latest/user/advanced/) will keep cookies persistent throughout the entire session, so even though a cookie is not provided in the first request, the session will use the cookie that Southwest sets for the next request. This can be verified with the following change to `lib/utils.py`:
```diff
                 response = session.get(url, headers=headers, params=info)
 
+            print(session.cookies)
+            continue
             if response.status_code == 200:
```

While cookies (or other parameters persistent across requests) may not be used to detect the script now, I wanted to bring up that point as a possibility that it may happen in the future.